### ISSUE TITLE
Marshaling map[string]interface{} values containing a pointer results in incorrect output

### ISSUE THREAD

**Issue by preethamrn** [goccy/go-json#261]
Sample test:
```go
func Test_string_pointer_marshal(t *testing.T) {    
    v := "value"    
    data := map[string]interface{}{        
        "data": &v,
    }
    var bufGoccy, bufNative bytes.Buffer    
    goccy.NewEncoder(&bufGoccy).Encode(data)    
    json.NewEncoder(&bufNative).Encode(data)
    assert.Equal(t, fmt.Sprintf("{\"data\":\"value\"}\n"), bufGoccy.String())    
    assert.Equal(t, fmt.Sprintf("{\"data\":\"value\"}\n"), bufNative.String())
}
```

```
expected: "{\"data\":\"value\"}\n" actual : "{\"data\":1.234452e-316}\n"
```

**Comment by preethamrn** [comment]
I was able to isolate the bug to this commit (994dc9ea9d57705f375065bbed394194d619907c) using git bisect.

**Comment by goccy** [comment]
Thank you for your great report. I will fix this issue with #262

**Comment by goccy** [comment]
I released this fix as v0.7.3 ! Thank you !!
